src/wallet/wallet.cpp

```diff
@@ -2834,103 +2834,111 @@ BlindingStatus CWallet::WalletBlindPSBT(PartiallySignedTransaction& psbtx) const
         }
 
         // The signature can't depend on witness contents, so these are technically not necessary to sign.
         // HOWEVER, as long as we're checking that values balance before signing, they are required.
         CTxOutWitness& outwit = tx.witness.vtxoutwit[i];
-        if (!output.range_proof.empty()) {
-            outwit.vchRangeproof = output.range_proof;
+        if (!output.m_value_rangeproof.empty()) {
+            outwit.vchRangeproof = output.m_value_rangeproof;
         }
-        if (!output.surjection_proof.empty()) {
-            outwit.vchSurjectionproof = output.surjection_proof;
+        if (!output.m_asset_surjection_proof.empty()) {
+            outwit.vchSurjectionproof = output.m_asset_surjection_proof;
         }
     }
 
-    // Stuff in the peg-in data
+    // Stuff in the peg-in and issuance data
     for (unsigned int i = 0; i < tx.vin.size(); ++i) {
         PSBTInput& input = psbtx.inputs[i];
-        if (input.value && input.peg_in_tx.which() != 0 && input.txout_proof.which() != 0 && !input.claim_script.empty() && !input.genesis_hash.IsNull()) {
+        CTxIn& txin = tx.vin[i];
+        CTxInWitness& txinwit = tx.witness.vtxinwit[i];
+        if (input.m_peg_in_value &&
+                input.m_peg_in_tx.which() > 0 &&
+                input.m_peg_in_txout_proof.which() > 0 &&
+                !input.m_peg_in_claim_script.empty() &&
+                !input.m_peg_in_genesis_hash.IsNull()) {
             CScriptWitness pegin_witness;
             if (Params().GetConsensus().ParentChainHasPow()) {
-                const Sidechain::Bitcoin::CTransactionRef& btc_peg_in_tx = boost::get<Sidechain::Bitcoin::CTransactionRef>(input.peg_in_tx);
-                const Sidechain::Bitcoin::CMerkleBlock& btc_txout_proof = boost::get<Sidechain::Bitcoin::CMerkleBlock>(input.txout_proof);
-                pegin_witness = CreatePeginWitness(*input.value, input.asset, input.genesis_hash, input.claim_script, btc_peg_in_tx, btc_txout_proof);
+                pegin_witness = CreatePeginWitness(*input.m_peg_in_value, Params().GetConsensus().pegged_asset, input.m_peg_in_genesis_hash, input.m_peg_in_claim_script, boost::get<Sidechain::Bitcoin::CTransactionRef&>(input.m_peg_in_tx), boost::get<Sidechain::Bitcoin::CMerkleBlock>(input.m_peg_in_txout_proof));
             } else {
-                const CTransactionRef& elem_peg_in_tx = boost::get<CTransactionRef>(input.peg_in_tx);
-                const CMerkleBlock& elem_txout_proof = boost::get<CMerkleBlock>(input.txout_proof);
-                pegin_witness = CreatePeginWitness(*input.value, input.asset, input.genesis_hash, input.claim_script, elem_peg_in_tx, elem_txout_proof);
+                input.m_peg_in_witness = CreatePeginWitness(*input.m_peg_in_value, Params().GetConsensus().pegged_asset, input.m_peg_in_genesis_hash, input.m_peg_in_claim_script, boost::get<CTransactionRef&>(input.m_peg_in_tx), boost::get<CMerkleBlock>(input.m_peg_in_txout_proof));
```

Comment:
I'm having trouble understanding this. In the PoW case we assign the pegin witness to `pegin_witness` and it is later put into the transaction; meanwhile we leave `input.m_peg_in_witness` alone. In the non-PoW case we assign the pegin witness to `input.m_peg_in_witness` and leave `pegin_witness` blank, so we then blank the pegin witness in the transaction.

If this is intended it at least needs a comment explaining the discrepancy between the PoW and non-PoW behavior.

Reply:
That's a mistake. Changed to set `pegin_witness`.